libcore/FileUtils.vala

```diff
@@ -881,6 +881,52 @@ namespace PF.FileUtils {
         }
     }
 
+    public string get_afp_target_uri (string standard_target_uri, string uri) {
+        /* For afp:// addresses the standard target uri has the user name stripped - we need to replace it */
+        if (Uri.parse_scheme (uri) == "afp") {
+            string origin_host, origin_filename;
+            if (get_afp_user_server_and_filename (uri, out origin_filename, out origin_host)) {
+                string target_host, target_filename;
+                if (get_afp_user_server_and_filename (standard_target_uri, out target_filename, out target_host)) {
+                    return "afp://" + Path.build_path (Path.DIR_SEPARATOR_S, origin_host, target_filename);
+                }
+            }
+        } else if (Uri.parse_scheme (standard_target_uri) == "afp") {
```

Comment:
Since we already check if the target URI scheme is `afp` before calling this function, we probably don't need to check again right? It's probably reasonable to assume that we would always check that target URI is `afp` in order to call this function right?

Reply:
Yes, I guess that is safe enough.  Replacing check with a comment.